### ISSUE TITLE
Add to FAQ: Why would I want the compiler to generate IAM policies

### ISSUE THREAD

**Issue by ShaiBer** [winglang/wing#2188]
### Feature Spec

Draft response:

TLDR reply: You don't have to let the compiler do everything by itself, you can customize with [compiler plugins](https://docs.winglang.io/blog/2023/02/17/plugins) that let you control the Terraform code.

Longer version:

I get where you're coming from, I guess using Wing will be a bit of a leap of faith for many developers as they would feel they lose control.

The way I see it though is that a higher level abstraction over the cloud, with all its difficulties, is the inevitable future. It will happen just like we managed to abstract single machines. I'm old enough to remember a time when we needed to write code for specific file systems of specific operation systems in order to save a file to the disk.

Today, programming languages and frameworks have successfully abstracted the file system, and every other part of the single computer. In the future, there should be little reason in most cases to target specific cloud storage solutions, let alone set up IAM policies manually in many cases where the needed policy can be deduced from the application code.

I am aware that this is a process that requires time, and we need to give more control over the finer details in the meantime. This is why we have the [compiler plugins](https://docs.winglang.io/blog/2023/02/17/plugins) system that allows you to customize the Terraform output.

### Use Cases

Answer this common question

### Component

Documentation
### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by ShaiBer** [comment]
#2329 solved it
